eventmesh-connectors/eventmesh-connector-http/src/main/java/org/apache/eventmesh/connector/http/source/protocol/impl/GitHubProtocol.java

```
@@ -0,0 +1,215 @@
+
+
+    /**
+     * Initialize the protocol.
+     *
+     * @param sourceConnectorConfig source connector config
+     */
+    @Override
+    public void initialize(SourceConnectorConfig sourceConnectorConfig) {
+        // Initialize the protocol
+        Map<String, String> extraConfig = sourceConnectorConfig.getExtraConfig();
+        this.secret = extraConfig.get(SECRET_KEY);
+        if (StringUtils.isBlank(this.secret)) {
+            throw new EventMeshException("The secret is required for GitHub protocol.");
+        }
+    }
+
+    /**
+     * Handle the protocol message for GitHub.
+     *
+     * @param route     route
+     * @param boundedQueue queue info
+     */
+    @Override
+    public void setHandler(Route route, BoundedConcurrentQueue<Object> boundedQueue) {
+        route.method(HttpMethod.POST)
+            .handler(BodyHandler.create())
+            .handler(ctx -> {
+                // Get the payload
+                String payload = ctx.body().asString(Constants.DEFAULT_CHARSET.toString());
+
+                // Get the headers
+                MultiMap headers = ctx.request().headers();
+                // validate the signature
+                String signature = headers.get(WebhookConstants.GITHUB_SIGNATURE_256);
+                if (BooleanUtils.isFalse(validateSignature(signature, payload, secret))) {
+                    String errorMsg = String.format("signature is invalid, please check the secret. received signature: %s", signature);
+                    log.error(errorMsg);
+
+                    WebhookResponse response = WebhookResponse.builder()
+                        .msg(errorMsg)
+                        .handleTime(LocalDateTime.now())
+                        .build();
+
+                    // Return 400 Bad Request
+                    ctx.response().setStatusCode(HttpResponseStatus.BAD_REQUEST.code())
+                        .send(JSON.toJSONString(response, Feature.WriteMapNullValue));
+                    return;
```

You can throw an exception here and let failureHandler to response.

---

Good idea, I will modify it

---

Solved at  https://github.com/apache/eventmesh/pull/4913/commits/b175fd22b91aab83b3adb22131306f0b0df29de3